### PR TITLE
Madison/soc

### PR DESCRIPTION
**Pull request by madisonmay**

### CODE REVIEW

File: finetune/base.py [hunk 1]
```diff
@@ -146,82 +135,81 @@ def _eval(self, *tensors, feed_dict):
...
+        fit_language_model_only = (Y is None)
+        arr_encoded = self._text_to_ids(*Xs)
+        labels = None if fit_language_model_only else Y
+        return self._training_loop(
+            arr_encoded,
+            Y=labels,
+            batch_size=batch_size, 
+        )
 
+    def _training_loop(self, arr_encoded, Y=None, batch_size=None):
```
Comment (benleetownsend) [inline finetune/base.py]: This has undone half of my fix for validation split problems. This leaves the problem that at small dataset sizes the only example(s) of a given class can be split into a validation set without any errors. If you're fine with that then you can ignore this.

File: finetune/classifier.py [hunk 2]
```diff
@@ -12,7 +16,7 @@ def featurize(self, X, max_length=None):
                            Providing more than `max_length` tokens as input will result in truncation.
         :returns: np.array of features of shape (n_examples, embedding_size).
         """
-        return self._featurize(X, max_length=max_length)
+        return super().featurize(X, max_length=max_length)
 
     def predict(self, X, max_length=None):
```
Comment (benleetownsend) [inline finetune/classifier.py]: This means that the methods change their interface from the base class?
